endpoints/remotehosts/remotehosts.py

```diff
@@ -1985,8 +1987,38 @@ def remote_image_manager(thread_name, remote_name, connection, image_max_cache_s
             }
     thread_logger(thread_name, "images[podman]:\n%s" % (endpoints.dump_json(images["podman"])), remote_name = remote_name, log_prefix = log_prefix)
 
+    if settings["rickshaw"]["quay"]["refresh-expiration"]["api-url"] is not None:
+        thread_logger(thread_name, "Found configuration information necessary to utilize the quay API to obtain image expiration", remote_name = remote_name, log_prefix = log_prefix)
+        thread_logger(thread_name, "Quay API URL: %s" % (settings["rickshaw"]["quay"]["refresh-expiration"]["api-url"]), remote_name = remote_name, log_prefix = log_prefix)
+
+        for image in images["podman"].keys():
+            image_parts = image.split(":")
+
+            get_request = requests.get(settings["rickshaw"]["quay"]["refresh-expiration"]["api-url"] + "/tag", params = { "onlyActiveTags": True, "specificTag": image_parts[1] })
+
+            query_log_level = "info"
+            if get_request.status_code != requests.codes.ok:
+                query_log_level = "warning"
+
+            thread_logger(thread_name, "Quay API query for %s returned %d" % (image, get_request.status_code), log_level = query_log_level, remote_name = remote_name, log_prefix = log_prefix)
+
+            if get_request.status_code == requests.codes.ok:
+                image_json = get_request.json()
+                if len(image_json["tags"]) == 1:
+                    images["quay"][image] = image_json["tags"][0]
+                else:
+                    thread_logger(thread_name, "Quay API query for %s found %d tags" % (image, len(image_json["tags"])), log_level = "warning", remote_name = remote_name, log_prefix = log_prefix)
+
+        thread_logger(thread_name, "images[quay]:\n%s" % (endpoints.dump_json(images["quay"])), remote_name = remote_name, log_prefix = log_prefix)
+    else:
+        thread_logger(thread_name, "Configuration information necessary to utilize the quay API to obtain image expiration timestamps is not available", remote_name = remote_name, log_prefix = log_prefix)
+
     image_expiration = endpoints.image_expiration_gmepoch()
-    thread_logger(thread_name, "Images created before %d will be considered expired" % (image_expiration), remote_name = remote_name, log_prefix = log_prefix)
+    thread_logger(thread_name, "Images evaludated by their expiration data will be considered expired if it is before %d" % (image_expiration), remote_name = remote_name, log_prefix = log_prefix)
+
+    expiration_weeks = int(settings["rickshaw"]["quay"]["image-expiration"].rstrip("w"))
```

Comment:
If I understand it right, this will be used ONLY if expiration cannot be retrieved from the registry, as a fallback.
What if the expiration is not defined in the rickshaw config for some reason ? Could not find a default value (previously 2w).

Reply:
At the moment there is a "loose" guarantee that the expiration is in the `rickshaw-settings.json` file -- meaning that the upstream config file has a value in it.  If a user, for some reason, deleted that value from the config file then this would blow up.  The `2w` default is the value that is in the upstream repo for this parameter in the `rickshaw-settings.json` file:

https://github.com/perftool-incubator/rickshaw/blob/01fbff69fd0a9f35e1e5caa952bbde09f36a52c2/rickshaw-settings.json#L26

The "real" solution that I intend on doing at some point is to create a JSON schema for the `rickshaw-settings.json` file -- just like we have for so many other JSON files -- and so there would be required parameters, such as this expiration setting, and data formats that would cause validation failures much earlier in the execution cycle if they are absent/incorrect.  I'm not exactly sure why we didn't do that from the beginning, it's likely an oversight on my part.